File: src/easy/how_many_walls.zig
const std = @import("std");
const expectEqual = std.testing.expectEqual;

pub fn howManyWalls(walls: u32, width: u32, height: u32) u32 {
    const total: u32 = height * width;
    if (total <= walls) {
        return @divFloor(walls, total);
    }
    return 0;
}
test "Determine how many m² can be painted" {
    try expectEqual(howManyWalls(100, 4, 5), 5);
    try expectEqual(howManyWalls(10, 15, 12), 0);
    try expectEqual(howManyWalls(41, 3, 6), 2);
    try expectEqual(howManyWalls(50, 11, 5), 0);
    try expectEqual(howManyWalls(1, 1, 1), 1);
}

File: src/easy/how_many_walls.rs
use std::ops::Div;
pub fn how_many_walls(walls: u16, width: u16, height: u16) -> u16 {
    let total = width * height;
    match total <= walls {
        true => walls.div(total),
        false => 0,
    }
}
#[cfg(test)]
mod tests {
    use super::how_many_walls;
    #[test]
    fn test_how_many_walls() {
        assert_eq!(how_many_walls(100, 4, 5), 5);
        assert_eq!(how_many_walls(10, 15, 12), 0);
        assert_eq!(how_many_walls(41, 3, 6), 2);
        assert_eq!(how_many_walls(50, 11, 5), 0);
        assert_eq!(how_many_walls(1, 1, 1), 1);
    }
}
